pkg/sql/sem/tree/expr.go

```diff
@@ -898,6 +898,10 @@ func (node *Array) Format(ctx *FmtCtx) {
 	ctx.WriteString("ARRAY[")
 	ctx.FormatNode(&node.Exprs)
 	ctx.WriteByte(']')
+	if len(node.Exprs) == 0 && ctx.HasFlags(FmtParsable) && node.typ != nil {
```

Comment:
Ah... one of these problems? Man wouldn't it be nice to come up with a proto-based serialization format soon.

Is this going to be a problem for `DArray` too? In that case we can fix it by adjusting the `AmbiguousFormat` method. I wonder if the caveats currently in that method need to apply to this too...